### DISCUSSION TITLE
Execution order of signal handlers

### DISCUSSION THREAD

**Discussion by khaihoan1** [celery/celery#9193] [Q&A]
Hi, 
If I register 2 handlers to task_postrun signal, what is the execution order of them?

```
@task_postrun.connect
def a():
     print('Logic a')

@task_postrun.connect
def b():
     print('Logic b')
```
I searched for the answer but couldn't find it. 
Thanks

**Comment by Enirsa** [comment]
Cmon bruh, you're a programmer or what? Test it out or check the source code for signal disptaching

**Reply by khaihoan1** [reply]
Yes, I did check it and I see which is registered first will be executed first. However, I faced some issues that made me feel uncertain. We have 2 handlers (1 gets connection from pool via SQLAlchemy and does something, this one is registered first, and the second one is to call Session.close(), which returns the connection to the pool). All task types are registered with the second handler, just 1 task type is registered with the 1st handler, and I see the error "QueuePool limit of size <x> overflow <y> reached, connection timed out, timeout <z>" on this type of task. That's why I asked this question.
